piff/kolmogorov_model.py

```diff
@@ -0,0 +1,311 @@
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: model_kolmogorov
+"""
+
+from __future__ import print_function
+import numpy as np
+
+from .model import Model
+from .star import Star, StarFit, StarData
+
+class Kolmogorov(Model):
+    """An extremely simple PSF model that just considers the PSF as a sheared Kolmogorov with
+    parameters for full-width-half-maximum `fwhm`, the applied shear `g1` and `g2`, and optionally,
+    centroid parameters `cenu` and `cenv`.
+    """
+    def __init__(self, fastfit=False, force_model_center=True, logger=None):
+        """A PSF modeled as a Kolmogorov profile.
+
+        :param fastfit:  Use HSM moments for fitting.  Approximate, but fast.  [default: True]
+        :param force_model_center: If True, PSF model centroid is fixed at origin and
+                            PSF fitting will marginalize over stellar position.  If False, stellar
+                            position is fixed at input value and the fitted PSF may be off-center.
+                            [default: True]
+        :param logger:  A logger object for logging debug info. [default: None]
+        """
+        self._fastfit = fastfit
+        self._force_model_center = force_model_center
+        # Params are [u, v], fwhm, g1, g2
+        self._nparams = 3 if force_model_center else 5
+
+        self.kwargs = {
+            'fastfit' : fastfit,
+            'force_model_center' : force_model_center
+        }
+
+    @staticmethod
+    def hsmfit(star):
+        """Compute the hsm moments for a given star.
+
+        :param star:    A Star instance
+
+        :returns: (flux, cenu, cenv, sigma, g1, g2, flag)
+        """
+        import galsim
+        image, weight, image_pos = star.data.getImage()
+        mom = image.FindAdaptiveMom(weight=weight, strict=False)
+
+        fwhm = mom.moments_sigma / 0.4519  # Magic number to convert sigma -> FWHM for a Kolmogorov
+        shape = mom.observed_shape
+        # These are in pixel coordinates.  Need to convert to world coords.
+        jac = image.wcs.jacobian(image_pos=image_pos)
+        scale, shear, theta, flip = jac.getDecomposition()
+        # Fix sigma
+        fwhm *= scale
+        # Fix shear.  First the flip, if any.
+        if flip:
+            shape = galsim.Shear(g1 = -shape.g1, g2 = shape.g2)
+        # Next the rotation
+        shape = galsim.Shear(g = shape.g, beta = shape.beta + theta)
+        # Finally the shear
+        shape = shear + shape
+
+        # Another magic number below to convert flux -> actual flux for Kolmogorov.
+        flux = mom.moments_amp / 0.9053
+
+        center = image.wcs.toWorld(mom.moments_centroid) - image.wcs.toWorld(image_pos)
```

Comment:
Probably a lot of things here can be backported to the Gaussian module.  In fact, pretty much the only thing that's actually different conceptually between Gaussian and Kolmogorov are the magic numbers in this method.  Probably suggests we should use a common parent class.  Something like "GSObject_Model" might work.  I'll give that a think...